src/Microsoft.AspNetCore.Mvc.Core/Internal/ObjectResultExecutor.cs

```diff
@@ -175,71 +182,57 @@ public virtual Task ExecuteAsync(ActionContext context, ObjectResult result)
                 throw new ArgumentNullException(nameof(formatters));
             }
 
-            // Check if any content-type was explicitly set (for example, via ProducesAttribute
-            // or URL path extension mapping). If yes, then ignore content-negotiation and use this content-type.
-            if (contentTypes.Count == 1)
+            var request = formatterContext.HttpContext.Request;
+            var acceptableMediaTypes = GetAcceptableMediaTypes(contentTypes, request);
+            IOutputFormatter selectedFormatter = null;
+            bool selectFormatterWithoutRegardingAcceptHeader = false;
```

Comment:
Noted; how does the team feel about using `default` to facilitate `var` for `null` values?